internal/appsec/waf/waf_test.go

```diff
@@ -406,6 +407,105 @@ func TestRunError(t *testing.T) {
 	}
 }
 
+func TestMetrics(t *testing.T) {
+	rules := `
+{
+  "version": "2.1",
+  "metadata": {
+    "rules_version": "1.2.7"
+  },
+  "rules": [
+	{
+	  "id": "valid-rule",
+	  "name": "Unicode Full/Half Width Abuse Attack Attempt",
+	  "tags": {
+		"type": "http_protocol_violation"
+	  },
+	  "conditions": [
+		{
+		  "parameters": {
+			"inputs": [
+			  {
+				"address": "server.request.uri.raw"
+			  }
+			],
+			"regex": "\\%u[fF]{2}[0-9a-fA-F]{2}"
+		  },
+		  "operator": "match_regex"
+		}
+	  ],
+	  "transformers": []
+	},
+	{
+	  "id": "missing-tags-1",
+	  "name": "Unicode Full/Half Width Abuse Attack Attempt",
+	  "conditions": [
+	  ],
+	  "transformers": []
+	},
+	{
+	  "id": "missing-tags-2",
+	  "name": "Unicode Full/Half Width Abuse Attack Attempt",
+	  "conditions": [
+	  ],
+	  "transformers": []
+	},
+	{
+	  "id": "missing-name",
+	  "tags": {
+		"type": "http_protocol_violation"
+	  },
+	  "conditions": [
+	  ],
+	  "transformers": []
+	}
+  ]
+}
+`
+	waf, err := NewHandle([]byte(rules))
+	require.NoError(t, err)
+	defer waf.Close()
+	t.Run("RulesetInfo", func(t *testing.T) {
+		rInfo := waf.RulesetInfo()
+		require.Equal(t, uint16(3), rInfo.Failed)
+		require.Equal(t, uint16(1), rInfo.Loaded)
+		require.Equal(t, 2, len(rInfo.Errors))
+		require.Equal(t, "1.2.7", rInfo.Version)
+		for key, arr := range rInfo.Errors {
```

Comment:
I think you can replace this with a clear `require.Equal()`:
```go
require.Equal(t, map[string]interface{}{
  "missing key 'tags'": []interface{}{"missing-tags-1", "missing-tags-2"}
  ...
}, rInfo.Errors)
```

Reply:
That's so much simpler, thanks